Pagination Options

jbrockopp opened the pull request:
@andygrunwald Let me know if you need anything else, thanks!

jbrockopp committed:
add ListOptions in support of pagination

jbrockopp committed:
add list options to get all incidents

andygrunwald commented:
Hey @jbrockopp, thanks for this.
The unit tests are failing right now. Would you mind to add tests for this as well?

jbrockopp committed:
add list options to rest of get all functions

jbrockopp committed:
update tests for list options

jbrockopp commented:
@andygrunwald I add the list options to the other GetAll() functions. I fixed the unit tests. I also added new unit tests for the GetAll() functions testing the list options functionality. 

Let me know if you need anything else, thanks!

andygrunwald commented:
Thank  you :)
